test_utilities/src/cloudflare.rs

```diff
@@ -1,15 +1,24 @@
 use crate::utilities::FuncTestsSecrets;
 use qovery_engine::dns_provider::cloudflare::Cloudflare;
-use qovery_engine::models::Context;
+use qovery_engine::models::{Context, Domain};
 
-pub fn dns_provider_cloudflare(context: &Context) -> Cloudflare {
+pub enum CloudflareDomain {
```

Comment:
Why Default is useful? I don't see any reasons where it's used instead for dropping an error. Why not using a string instead in dns_provider_cloudflare?

Reply:
I renamed it to be `ClusterDomain` instead.

- Default allows to say to the engine to let Cloudflare and DNS providers to use their default domain given via secrets or whatever, whereas custom indicate to use the custom one, agin, a bit more clear when you see it in code later on.

- We should try to avoid setting everything inside a string as much as we can and leverage types :)

- As of why it's better using an enum rather than a &str is that types carry information and it's easier to maniupulate and know what it should represent.